doc/sphinx/README.rst

```diff
@@ -64,7 +64,26 @@ The signatures of most objects can be written using a succinct DSL for Coq notat
   an alternative, indicating than one of multiple constructs can be used
 
 ``%{``, ``%}``, ``%|``
-  an escaped character (rendered without the leading ``%``).  In most cases, escaping is not necessary.  In particular, the following expressions are all parsed as plain text, and do not need escaping: ``{ xyz }``, ``x |- y``.  But the following escapes *are* needed: ``{| a b %| c | d }``, ``all: %{``.  (We use ``%`` instead of the usual ``\`` because you'd have to type ``\`` twice in your reStructuredText file.)
+  an escaped character (rendered without the leading ``%``).  We use ``%``
+  instead of the usual ``\`` because reStructuredText would require using
+  the awkward double escape ``\\``.
+
+  - Use it to escape grammar literal strings that are the same as metasyntax,
+  such as ``{``, ``|``, ``}``, ``{|`` and ``{+,``.  (While this is optional for
+  ``|`` and ``{ ... }`` outside of ``{| ... }``, always using the escape
+  requires less thought.)
```

Comment:
I don't like this part too much.  Is it useful to explain that ``{+,`` needs to be escaped? And do we really want plenty of `%` signs to pop up where unnecessary?

Reply:
I took out `{+,`.   The grammar does use the literal `{|`.

It's much easier to describe and follow an unconditional rule, e.g. "always escape `{`, `|` and `}`".  `%` behaves much differently from the usual backslash escape and it has several exceptions--it operates on tokens, `%%` stands for itself, etc.

I think the number of extra `%` would be small.  For example, there are only 18 places in the .rst files that match the regex `\.\..*{ `, which finds the relevant directives.

Also, BTW, `name1 += }` and `name1 += {` give parse errors, though AFAIK that's not an issue for our grammar.